### ISSUE TITLE
Hillshading accent color

### ISSUE THREAD

**Issue by utagawal** [openlayers/ol-mapbox-style#725]
For the accent color implementation in hillshading , maybe usefull to take inspiration from here : src/shaders/hillshade.fragment.glsl
from https://github.com/mapbox/mapbox-gl-js/pull/5286/files#diff-a826d96571ff241535a6bac52167defaabc13c91883e2529cf84608c70ca9b79
(but it's using webGL)

**Comment by utagawal** [comment]
In the current code - didn't find also where the highlight and shadow colors colors are gathered and applied.


It should probably be in apply.js around 
``
const hillshadeLayer = setupHillshadeLayer(...


For the accent color implementation the example code from maplibre is extracted below :

```
// we multiply the slope by a scale factor based on the cosin of the pixel's approximate latitude
    float scaleFactor = cos(radians((u_latrange[0] - u_latrange[1]) * (1.0 - v_pos.y) + u_latrange[1]));
    float slope = atan(1.25 * length(deriv)) / scaleFactor;
    float aspect = deriv.x != 0.0 ? atan(deriv.y, -deriv.x) : PI / 2.0 * (deriv.y > 0.0 ? 1.0 : -1.0);

    float intensity = u_light.x;
    // We add PI to make this property match the global light object, which adds PI/2 to the light's azimuthal
    // position property to account for 0deg corresponding to north/the top of the viewport in the style spec
    // and the original shader was written to accept (-illuminationDirection - 90) as the azimuthal.
    float azimuth = u_light.y + PI;
    float polar = u_light.z;

    float accent = cos(slope);
    vec4 accent_color = intensity * clamp((1.0 - accent) * 2.0, 0.0, 1.0) * u_accent;
    float shade = abs(mod((aspect + azimuth) / PI + 0.5, 2.0) - 1.0);
    vec4 shade_color = mix(u_shadow, u_highlight, shade) * slope * sin(polar) * mix(0.0, 2.0, intensity);
    gl_FragColor = accent_color * (1.0 - shade_color.a) + shade_color;
```

**Comment by ahocevar** [comment]
The current implementation uses the same technique as the https://openlayers.org/en/latest/examples/shaded-relief.html example. No colors can be configured. To make the result look good, a version of OpenLayers that includes https://github.com/openlayers/openlayers/pull/14282 is needed. That was a big change and makes a big difference.

If someone can provide a pull request that implements the color options, it would be greatly appreciated.

**Comment by ahocevar** [comment]
Implementation hints: The rendering transformation takes place in https://github.com/openlayers/ol-mapbox-style/blob/f5c134c03bebcb8ac3b444a5b990841c6aa788d6/src/shaders.js. The `data` passed to that function is calculated in https://github.com/openlayers/ol-mapbox-style/blob/f5c134c03bebcb8ac3b444a5b990841c6aa788d6/src/apply.js#L819-L846. This is where the `*-color` options would have to be added to the `data` object.

**Comment by utagawal** [comment]
Merci beaucoup Virgil ! If you have some additional bandwidth we are also looking at the line-pattern implementation https://github.com/openlayers/ol-mapbox-style/issues/713

**Comment by V-Roger** [comment]
@ahocevar Hey there, I took the liberty of jumping in to rescue @utagawal :)

I couldn't find a contribution guide here, so hope the PR is alright. Feel free to request changes!

**Comment by utagawal** [comment]
Should be ready to be published in a stable release version

**Comment by utagawal** [comment]
@ahocevar We'd like to use the pull request on our website, any chance you could include it in a new release version soon ?

**Comment by ahocevar** [comment]
@utagawal Please be patient. I have a lot of work that pays my bills to finish before I can maintain this and other libraries in my free time. Speaking of, I'm grateful for your one-time sponsoring a while ago. I used those funds for adding the first iteration of the hillshading capability, which was quite an involved change that also required changes to the OpenLayers core library.

**Comment by ahocevar** [comment]
@utagawal I published v9.3.0.

**Comment by utagawal** [comment]
Many thanks @ahocevar
